torch/csrc/jit/codegen/cuda/fusion.h

```diff
@@ -261,6 +277,10 @@ class TORCH_CUDA_API Fusion : public IRInputOutput {
 
   // Map a subset of values to the lowered equivalent (ex. sizes)
   std::unordered_map<Val*, Val*> values_map_;
+
+  // values for launch configuration:
+  // compatible_flag, BlockDim.x/y/z, GridDim.x/y/z, shared_memory,
+  std::unordered_map<LaunchConfigType, Val*> launch_configs_;
```

Comment:
1. Can we have the launch config outside, and independent of the Fusion? 
2. map to `const Val*`? (or even better, directly to `Int::ScalarType`)

Reply:
Regarding 1. This is the part I'm still thinking about. I wouldn't mind having this outside of the `Fusion` with all arith nodes. (a separate `Fusion` with only the inputs & config logic nodes); But I don't know if we want to have a map of pointer to nodes, whose lifetime is owned by a separate `fusion` object. I could have them bounded together by a wrapper object tho.
Curious on the motivation of the separation? Are we worried about overhead?

2. `const` would make sense, but ExpressionEvaluator::evaluate takes `Val*` so we have to const_cast the specifier away, maybe we can update the function signature? https://github.com/csarofeen/pytorch/blob/481afe10e5d967f7800cde69cde70c82dde50c46/torch/csrc/jit/codegen/cuda/expr_evaluator.h#L50-L52